### PR TITLE
fix(dashboard): Fix filter_scopes override on json editor

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
Since `filter_scopes` key overrides by the redux state, hard-coded `filter_scopes` value from json editor ignored.

https://github.com/apache/superset/blob/75e6a04269bf73c0c7160290333ded4e63421a4b/superset-frontend/src/dashboard/actions/dashboardState.js#L356-L362

This commit adds the redux updateDashboardFiltersScope logic from the json editor.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

- Before

https://user-images.githubusercontent.com/1392866/195411595-776d3c4a-209e-4b3d-96dc-ff1b1cd1f92d.mov

- After

https://user-images.githubusercontent.com/1392866/195411604-333489b7-94db-43a7-9c7f-928e37f2be14.mov

### TESTING INSTRUCTIONS
Go to a dashboard has a filter box
Open "Edit Properties" and then change the `filter_scopes` value from JSON editor
Check the updates from "set filter mapping" modal

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

cc: @ktmud

### CODE REVIEW

**Review by stephenLYZ — APPROVED** [review]
LGTM

**Review by ktmud — APPROVED** [review]
LGTM but it seems you have a conflict

**Comment by rusackas** [comment]
Thanks for the contribution @justinpark - holler if anyone here can be of help with the conflict resolution!

File: superset-frontend/src/dashboard/components/Header/index.jsx [hunk 1]
```diff
@@ -45,6 +45,8 @@ import PublishedStatus from 'src/dashboard/components/PublishedStatus';
 import UndoRedoKeyListeners from 'src/dashboard/components/UndoRedoKeyListeners';
 import PropertiesModal from 'src/dashboard/components/PropertiesModal';
 import { chartPropShape } from 'src/dashboard/util/propShapes';
+import { getDashboardFilterKey } from 'src/dashboard/util/getDashboardFilterKey';
+import { UserWithPermissionsAndRoles } from 'src/types/bootstrapTypes';
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/Header/index.jsx:49]: Builds are failing because `UserWithPermissionsAndRoles` is not being used

**Pull request closed without merging**